compiler/noirc_evaluator/src/ssa/opt/mem2reg/block.rs

```diff
@@ -256,6 +240,13 @@ impl Block {
 
     pub(super) fn keep_last_load_for(&mut self, address: ValueId) {
         self.last_loads.remove(&address);
-        self.for_each_alias_of(address, |block, alias| block.last_loads.remove(&alias));
+
+        if let Some(expr) = self.expressions.get(&address) {
+            if let Some(aliases) = self.aliases.get(expr) {
```

Comment:
Here I didn't use `get_aliases_for_value` because we'd have to clone the return value to change `self.last_loads`. So this change avoids one clone.